raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,563 @@
+        // participant, used to compute the net balance on settlement.
+        uint256 transferred_amount;
+
+        // Value used to order transfers and only accept the latest on calls to
+        // update, this will only be relevant after either #182 or #293 is
+        // implemented.
+        uint64 nonce;
+
+        // This value is set to True after the channel has been opened.
+        // This is an efficient way to mark the channel participants without doing a deposit.
+        // This is uint8 and it gets packed with the nonce.
+        bool initialized;
+
+        // A mapping to keep track of locks that have been withdrawn.
+        mapping(bytes32 => bool) withdrawn_locks;
+    }
+
+    struct Channel {
+        // We also use the settle_timeout to verify that
+        // the channel is open
+        uint settle_timeout;
+        mapping(address => Participant) participants;
+    }
+
+    struct ClosingRequest {
+        address closing_address;
+
+        // Block number at which the settlement window ends.
+        uint settle_block_number;
+    }
+
+    /*
+     *  Events
+     */
+
+    event ChannelOpened(
+        uint channel_identifier,
+        address participant1,
+        address participant2,
+        uint settle_timeout
+    );
+
+    event ChannelNewBalance(uint channel_identifier, address participant, uint balance);
+
+    event ChannelClosed(uint channel_identifier, address closing_address);
+
+    event ChannelUnlocked(uint channel_identifier, address participant, uint transferred_amount);
+
+    event TransferUpdated(uint channel_identifier, address participant);
+
+    event ChannelSettled(uint channel_identifier);
+
+    /*
+     * Modifiers
+     */
+
+    modifier isClosed(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number > 0);
+        _;
+    }
+
+    modifier stillTimeout(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number > block.number);
+        _;
+    }
+
+    modifier timeoutOver(uint channel_identifier) {
+        require(closing_requests[channel_identifier].settle_block_number <= block.number);
+        _;
+    }
+
+    // Note: we use the settleTimeout to check if the channel is open. It must always be > 0.
+    modifier settleTimeoutValid(uint timeout) {
+        require(timeout >= 6 && timeout <= 2700000);
+        _;
+    }
+
+    /*
+     *  Constructor
+     */
+
+    function TokenNetwork(address _token_address) public {
+        require(_token_address != 0x0);
+        require(contractExists(_token_address));
+
+        token = Token(_token_address);
+
+        // Check if the contract is indeed a token contract
+        require(token.totalSupply() > 0);
+    }
+
+    /*
+     *  Public functions
+     */
+
+    /// @notice Opens a new channel between `participant1` and `participant2`.
+    /// Can be called by anyone.
```

Comment:
The specs don't say if this should be callable by anyone. (however, I don't see issues with that)

Reply:
Discussed this in the spec doc (comments). But it's good to think it over again. 
I think open + deposit can safely be callable by anyone.